### PR TITLE
fix: Better certificate error messages

### PR DESCRIPTION
**Pull request by ffflorian**

### CODE REVIEW

File: electron/js/certutils.js [hunk 1]
```diff
@@ -77,19 +77,51 @@ module.exports = {
...
+            algorithmIDCheck,
+            algorithmParamCheck
           );
+
           return arr;
-        }, []).every((val) => val !== false);
+        }, []).every(value => value !== false);
+
+        if (errorMessages.length > 0) {
+          result.errorMessage = errorMessages.join(' ');
```
Comment (gregor) [inline electron/js/certutils.js]: Let's join them by a new line for easier readability

File: electron/main.js [hunk 2]
```diff
@@ -520,9 +520,10 @@ class ElectronWrapperInit {
 
             if (certutils.hostnameShouldBePinned(hostname)) {
               const pinningResults = certutils.verifyPinning(hostname, certificate);
+
               for (const result of Object.values(pinningResults)) {
                 if (result === false) {
-                  console.error(`Certutils verification failed for ${hostname}: ${result} is false`);
+                  console.error(`Certutils verification failed for "${hostname}": ${pinningResults.errorMessage}`);
```
Comment (gregor) [inline electron/main.js]: How about moving the error messages into new lines.

File: electron/main.js [hunk 3]
```diff
@@ -520,9 +520,10 @@ class ElectronWrapperInit {
 
             if (certutils.hostnameShouldBePinned(hostname)) {
               const pinningResults = certutils.verifyPinning(hostname, certificate);
+
               for (const result of Object.values(pinningResults)) {
                 if (result === false) {
```
Comment (lipis) [inline electron/main.js]: `!result`

File: electron/js/certutils.js [hunk 4]
```diff
@@ -77,19 +77,51 @@ module.exports = {
...
-            algorithmID === publicKey.algoid,
-            algorithmParam === publicKey.algparam
+            fingerprintCheck,
+            algorithmIDCheck,
+            algorithmParamCheck
           );
+
           return arr;
-        }, []).every((val) => val !== false);
+        }, []).every(value => value !== false);
```
Comment (lipis) [inline electron/js/certutils.js]: `Boolean(value)`

**Pull request merged**